cerbero/tools/depstracker.py

```diff
@@ -85,7 +104,8 @@ def __init__(self, platform, prefix):
         self.lister = self.BACKENDS[platform]()
 
     def list_deps(self, path):
-        deps = self.lister.list_deps(self.prefix, path)
+        loop = asyncio.get_event_loop()
+        deps = loop.run_until_complete(self.lister.async_list_deps(self.prefix, path))
```

Comment:
What is the point on converting these functions to async if they are not being parallelized at any point?

Reply:
Mainly to get rid of this warning:
Exception ignored when trying to write to the signal wakeup fd:
BlockingIOError: [Errno 11] Resource temporarily unavailable
which needs python >= 3.8 to be corrected

Reply:
Just to understand, who is raising this error, shell.check_call ?
I get the interest of making the API async, but it's useless has if  DepsTracker.list_deps is not async itself, the DepsTracker will still be synchronous.
It's redundant to prefix functions with async_, this is normally only used when there are 2 variants of the function async and not async.

Reply:
Yes this error is raised by shell.check_call().
In fact all subprocess methods are raising it when called more than 300 times/sec (which is the case when doing a lot of strip or ldd while building a package). This warning is fixed in Python >= 3.8 but for now it pollutes a lot the logs and may hide other warnings more important.

In the upstream version of Cerbero the whole system goes towards asynchronous calls for system calls, so I begun to adapt it here to have it ready for future usage and get rid of the warning at the same time.

I didn't want to spend time modifying all the callers, so it's true that for now, having it asynchronous has very few interest from a performance point of view.